roles/translator/src/proxy/bridge.rs

```diff
@@ -291,38 +320,49 @@ impl Bridge {
     /// `SetNewPrevHash` `job_id`, an error has occurred on the Upstream pool role and the
     /// connection will close.
     fn handle_new_extended_mining_job(self_: Arc<Mutex<Self>>) {
+        let (tx_sv1_notify, rx_sv2_new_ext_mining_job, tx_status) = self_
+            .safe_lock(|s| {
+                (
+                    s.tx_sv1_notify.clone(),
+                    s.rx_sv2_new_ext_mining_job.clone(),
+                    s.tx_status.clone(),
+                )
+            })
+            .unwrap();
         debug!("Starting handle_new_extended_mining_job task");
         task::spawn(async move {
-            let tx_status = self_.safe_lock(|s| s.tx_status.clone()).unwrap();
             loop {
                 // Receive `NewExtendedMiningJob` from `Upstream`
-                let rx_sv2_new_ext_mining_job = self_
-                    .safe_lock(|r| r.rx_sv2_new_ext_mining_job.clone())
-                    .unwrap();
                 let sv2_new_extended_mining_job: NewExtendedMiningJob =
-                    rx_sv2_new_ext_mining_job.clone().recv().await.unwrap();
+                    handle_result!(tx_status, rx_sv2_new_ext_mining_job.clone().recv().await);
                 debug!(
                     "handle_new_extended_mining_job job_id: {:?}",
                     &sv2_new_extended_mining_job.job_id
                 );
-                let _ = self_
+                let res = self_
                     .safe_lock(|s| {
                         s.channel_factory.on_new_extended_mining_job(
                             sv2_new_extended_mining_job.as_static().clone(),
                         )
                     })
-                    .unwrap();
+                    .map_err(|_| PoisonLock);
+                let res = handle_result!(tx_status, res);
+                handle_result!(tx_status, res);
 
                 // If future_job=true, this job is meant for a future SetNewPrevHash that the proxy
                 // has yet to receive. Insert this new job into the job_mapper .
                 if sv2_new_extended_mining_job.future_job {
-                    self_
+                    let res = self_
                         .safe_lock(|s| s.future_jobs.push(sv2_new_extended_mining_job.clone()))
-                        .unwrap();
+                        .map_err(|_| PoisonLock);
+                    handle_result!(tx_status, res);
 
                 // If future_job=false, this job is meant for the current SetNewPrevHash.
                 } else {
-                    let last_p_hash = self_.safe_lock(|s| s.last_p_hash.clone()).unwrap();
+                    let last_p_hash = self_
+                        .safe_lock(|s| s.last_p_hash.clone())
+                        .map_err(|_| PoisonLock);
+                    let last_p_hash = handle_result!(tx_status, last_p_hash);
```

Comment:
Can this one be removed? If not can you leave a comment as to why handle_result! is being called twice?